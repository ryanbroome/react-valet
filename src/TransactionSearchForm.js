import React, { useState, useContext } from "react";

import { FormGroup, Form, Col, Input, Button } from "reactstrap";
import ThemeContext from "./ThemeContext";

const TransactionSearchForm = ({ search, resetList }) => {
  const { userDetail } = useContext(ThemeContext);

  const INITIAL_STATE = {
    mobile: "",
  };

  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData({
      ...formData,
      [name]: value,
    });
  };

  const handleSubmit = (evt) => {
    evt.preventDefault();
    search(userDetail.locationId, formData.mobile);
    setFormData({ mobile: "" });
  };

  return (
    <Form onSubmit={handleSubmit}>
      {Object.keys(INITIAL_STATE).map((val, idx) => (
        <Col
          md={10}
          key={`Col-${idx}`}>
          <FormGroup key={`FormGroup-${val}`}>
            <Input
              id={val}
              key={`Input-${val}`}
              type="text"
              name={val}
              placeholder={val.toLowerCase()}
              value={formData[val]}
              onChange={handleChange}></Input>
          </FormGroup>
        </Col>
      ))}

      <Button color="primary">Search</Button>
      <Button
        color="danger"
        onClick={resetList}>
        Reset
      </Button>
    </Form>
  );
};

export default TransactionSearchForm;
