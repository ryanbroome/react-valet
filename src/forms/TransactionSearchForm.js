import React, { useState, useContext } from "react";

import { Form, Col, Input, Button, Row, Label, InputGroup } from "reactstrap";
import UserContext from "../auth/UserContext";

const TransactionSearchForm = ({ search, reset, lostKeys }) => {
  const { userDetail } = useContext(UserContext);

  const INITIAL_STATE = {
    mobile: "",
  };

  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData({
      ...formData,
      [name]: value,
    });
  };

  const handleSubmit = (evt) => {
    evt.preventDefault();
    search(userDetail.locationId, formData.mobile);
    setFormData({ mobile: "" });
  };

  return (
    <Form
      onSubmit={handleSubmit}
      className="fixed-bottom bg-dark"
      zindex="-1">
      <Row className="row-cols-lg-auto g-3 align-items-end">
        <Col>
          <Label
            className="visually-hidden"
            htmlFor="mobile"></Label>
          <InputGroup>
            <Input
              id="mobile"
              type="text"
              name="mobile"
              placeholder="mobile"
              value={formData.mobile}
              onChange={handleChange}
            />
            <Button color="success">Find Mobile</Button>
          </InputGroup>
        </Col>
      </Row>
      <Row> {` =========================================================`}</Row>
      <Row>
        <Col>
          <Button
            color="success"
            onClick={() => {
              reset(userDetail.locationId, "parked");
            }}>
            Show All
          </Button>
        </Col>

        <Col>
          <Button
            color="danger"
            onClick={() => {
              lostKeys(userDetail.locationId, userDetail.id);
            }}>
            Lost Keys
          </Button>
        </Col>
      </Row>
    </Form>
  );
};

export default TransactionSearchForm;
