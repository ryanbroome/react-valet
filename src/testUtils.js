import React from "react";
import UserContext from "./auth/UserContext";
import App from "../src/App";

const MockUserContext = ({ children }) => {
  const mockToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InRlc3R1c2VyIiwibG9jYXRpb25JZCI6MSwiaXNBZG1pbiI6ZmFsc2UsImlhdCI6MTcxMzA3NTM4Mn0.dXlnawfzk6wlsgpouoSp58QLvlOOSJFC03iq_FXab-I";

  const mockUserDetail = {
    userId: 1,
    vehicleId: 1,
    locationId: 1,
    transactionId: 1,
    ticketNum: "1",
    mobile: "5553334444",
    color: "red",
    make: "honda",
    username: "testuser",
  };

  const mockLocation = {
    sitename: "testSitename",
  };

  const mockVehicle = {
    ticketNum: 1,
    vehicleStatus: `parked`,
    mobile: "1113334444",
    color: "testColor",
    make: "testMake",
    damages: "testDamages",
    notes: "testNotes",
  };

  const mockTransaction = {
    userId: 1,
    vehicleId: 1,
    locationId: 1,
  };

  return <UserContext.Provider value={{ token: mockToken, userDetail: mockUserDetail, location: mockLocation, vehicle: mockVehicle, transaction: mockTransaction, update: App.update, register: App.register }}>{children}</UserContext.Provider>;
};

export { MockUserContext };
