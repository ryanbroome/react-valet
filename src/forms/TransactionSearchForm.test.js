import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import TransactionSearchForm from "./TransactionSearchForm";
import { MemoryRouter } from "react-router";
import { MockUserContext } from "../testUtils";

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <MockUserContext children={<TransactionSearchForm />}></MockUserContext>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <MockUserContext children={<TransactionSearchForm />}></MockUserContext>
    </MemoryRouter>
  );
});
