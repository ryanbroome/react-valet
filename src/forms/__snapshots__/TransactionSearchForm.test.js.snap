// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form
      class=""
    >
      <button
        class="btn btn-primary"
      >
        Find
      </button>
      <div
        class="col-md-10"
      >
        <div
          class="position-relative form-group"
        >
          <input
            class="form-control"
            id="mobile"
            name="mobile"
            placeholder="mobile"
            type="text"
            value=""
          />
        </div>
      </div>
    </form>
    <button
      class="btn btn-warning"
      type="button"
    >
      Show All
    </button>
    <button
      class="btn btn-danger"
      type="button"
    >
      Lost Keys
    </button>
  </div>
</DocumentFragment>
`;
